api/handlers/exporter.go

```diff
@@ -59,35 +57,34 @@ func (e *Exporter) Decideds(w http.ResponseWriter, r *http.Request) error {
 
 	response.Data = []*ParticipantResponse{}
 
-	qbftStores := make(map[convert.RunnerRole]qbftstorage.QBFTStore, len(request.Roles))
 	for _, role := range request.Roles {
-		runnerRole := casts.BeaconRoleToConvertRole(spectypes.BeaconRole(role))
-		storage := e.QBFTStores.Get(runnerRole)
-		if storage == nil {
-			return api.Error(fmt.Errorf("role storage doesn't exist: %v", role))
-		}
-
-		qbftStores[runnerRole] = storage
-	}
-
-	for _, role := range request.Roles {
-		runnerRole := casts.BeaconRoleToConvertRole(spectypes.BeaconRole(role))
-		qbftStore := qbftStores[runnerRole]
+		participanStore := e.ParticipantStores.Get(spectypes.BeaconRole(role))
 
 		for _, pubKey := range request.PubKeys {
-			msgID := convert.NewMsgID(e.DomainType, pubKey, runnerRole)
 			from := phase0.Slot(request.From)
 			to := phase0.Slot(request.To)
 
-			participantsList, err := qbftStore.GetParticipantsInRange(msgID, from, to)
+			participantsRange, err := participanStore.GetParticipantsInRange(spectypes.BeaconRole(role), spectypes.ValidatorPK(pubKey), from, to)
 			if err != nil {
 				return api.Error(fmt.Errorf("error getting participants: %w", err))
 			}
 
-			if len(participantsList) == 0 {
+			if len(participantsRange) == 0 {
 				continue
 			}
 
+			var participantsList []qbftstorage.Participation
```

Comment:
```suggestion
			participantsList := make([]qbftstorage.Participation, 0, len(participantsRange))
```
to save some memory allocations